src/lib/homeobject_impl.hpp

```diff
@@ -21,15 +16,32 @@ namespace homeobject {
 inline bool operator<(ShardInfo const& lhs, ShardInfo const& rhs) { return lhs.id < rhs.id; }
 inline bool operator==(ShardInfo const& lhs, ShardInfo const& rhs) { return lhs.id == rhs.id; }
 
+struct ShardInfoExt {
```

Comment:
How about just:
```
struct ShardInfoExt : public ShardInfo {
```
Then we can change `_get_shard` to return a `Result<ShardInfoExt>` and get rid of `Shard`? (or use `Shard` instead of `ShardInfoExt`; but just one new struct i think is needed, not two).

Reply:
ok, will do that